cmd/cluster/aws/create.go

```diff
@@ -77,10 +85,89 @@ func applyPlatformSpecificsValues(ctx context.Context, exampleOptions *apifixtur
+		} else {
+			return err
+		}
+	} else {
+		opts.Log.Info("Found hosted cluster infrastructure", "namespace", opts.Namespace, "name", opts.InfraID)
+		if hci.Spec.SubDomain != opts.Name {
+			return fmt.Errorf(fmt.Sprintf("The HostedClusterInfrastructure.Spec.SubDomain must equal the cluster name\nCluster name: %s, Sub-Domain: %s", opts.Name, hci.Spec.SubDomain))
+		}
+		psStatus, infraStatus, iamStatus := hostedclusterinfrastructure.GetInfraIamConditionStatus(&hci)
+		if !psStatus || !infraStatus || !iamStatus {
+			return fmt.Errorf("the hosted cluster infrastructure resource is not configured as expected\n- provider-secret  : %t\n- infra-as-expected: %t\n- iam-as-expected  : %t",
+				psStatus, infraStatus, iamStatus)
+		}
+		//Set a finalizer on HCI
+		origHCI := hci.DeepCopy()
+		controllerutil.AddFinalizer(&hci, core.DestroyClusterFinalizer)
+		if hci.ObjectMeta.Labels == nil {
+			hci.SetLabels(map[string]string{})
+		}
+		hci.ObjectMeta.Labels[hostedClusterLabel] = opts.Name
+		if err := client.Patch(ctx, &hci, crclient.MergeFrom(origHCI)); err != nil {
+			return err
+		}
+
+		if opts.ServiceCIDR == "" {
+			exampleOptions.ServiceCIDR = "172.31.0.0/16"
+		}
+		if opts.ClusterCIDR == "" {
+			exampleOptions.ClusterCIDR = "10.132.0.0/14"
+		}
+		if opts.ControlPlaneAvailabilityPolicy == "" {
+			exampleOptions.ControlPlaneAvailabilityPolicy = "SingleReplica"
+		}
+		infra = &awsinfra.CreateInfraOutput{
+			Name:            opts.Name,
+			Region:          hci.Spec.Platform.AWS.Region,
+			BaseDomain:      hci.Spec.BaseDomain,
+			InfraID:         hci.Spec.InfraID,
+			MachineCIDR:     hci.Status.Infrastructure.Networking.MachineNetwork[0].CIDR.String(),
+			VPCID:           hci.Status.Infrastructure.Platform.AWS.VPC,
+			SecurityGroupID: *hci.Status.Infrastructure.Platform.AWS.SecurityGroups[0].ID,
+			PublicZoneID:    hci.Status.Infrastructure.DNS.PublicZoneID,
+			PrivateZoneID:   hci.Status.Infrastructure.DNS.PrivateZoneID,
+		}
+		infraID = hci.Spec.InfraID
+		infra.Zones = []*awsinfra.CreateInfraOutputZone{}
+		for _, zone := range hci.Status.Infrastructure.Platform.AWS.Zones {
+			newZone := &awsinfra.CreateInfraOutputZone{
+				Name:           zone.Zone,
+				SubnetID:       *zone.Subnet.ID,
+				PublicSubnetID: *zone.PublicSubnet.ID,
+			}
+			infra.Zones = append(infra.Zones, newZone)
+		}
+		iamInfo = &awsinfra.CreateIAMOutput{
+			Region:      hci.Spec.Platform.AWS.Region,
+			ProfileName: hci.Status.Infrastructure.Platform.AWS.InstanceProfile,
+			InfraID:     hci.Spec.InfraID,
+			IssuerURL:   hci.Status.Infrastructure.IssuerURL,
+			Roles:       *hci.Status.Infrastructure.Platform.AWS.RolesRef.DeepCopy(),
+
+			//KMSKeyARN: ,
+			//KMSProviderRoleARN: ,
+		}
+		opts.AWSPlatform.AdditionalTags = *hostedclusterinfrastructure.StringTagList(hci.Spec.Platform.AWS.ResourceTags)
+	}
 	if len(opts.InfrastructureJSON) > 0 {
```

Comment:
After getting the info from the HCI here now we transparently discarded and fetch it from the json, is that expected?

I'd expect useHCIResource to be a CLI choice analogous we have `cmd.PersistentFlags().StringVar(&opts.InfrastructureJSON, "infra-json", opts.InfrastructureJSON, "Path to file containing infrastructure information for the cluster. If not specified, infrastructure will be created")`

Reply:
Do you feel you need this for day zero, or it can be added next week.

Reply:
no longer applies